partially fixes #165 - created navbar, sidebars for storelisting page

djmgit opened the pull request:
Partially fixes issue #165. Created navigation bar, left side bar
for categores, and right bar for similar app suggestion. Implemented
deeplinks for category filtering.
A url of form http://apps.loklak.org/#<category_name> will show only
the apps belonging to the mentioned category.
Storelisting page will have the following url.
http://apps.loklak.org/details.html?q=<app_name>
The page will show all the informaton listed for given app (app_name)
The category to which the selected app belongs will be hghlighted.
Clicking on any category name will take the user to starting page
showing only the filtered apps which belong to the selected category.
The middle section (presently empty) will hold the app information.

I have:
- [x] There is a corresponding issue for this pull request.
- [x] Mentioned the Issue number in the pull request commit message `Fixes #<number> commit message`
- [x] There is only strictly only one commit per issue.

### For the reviewers
I have:
- [ ] Reviewed this pull request by an authorized contributor.
- [ ] The reviewer is assigned to the pull request.

djmgit commented:
Test link : loklakapps.surge.sh/details?q=whatsTrending

djmgit commented:
@kavithaenair @SKrPl open for review

mariobehling commented:
I see a big white space on your test link. How this supposed to look like?
![screenshot from 2017-06-02 18-09-14](https://cloud.githubusercontent.com/assets/1583873/26734523/ab73f330-47be-11e7-8860-495fa69b16e4.png)

djmgit commented:
mariobehling as I have mentioned here - https://github.com/fossasia/apps.loklak.org/issues/165 the middle section which is presently blank now will contain the selected app information, it will be loadded dynamically from app.json file of the selected app, so that the users will no how to use the app, the developer of the app will have to provide all the info in app.json of the app. I will be creating a tutorial page where everything will be mentioned about creating a an (part of my GSOC project)

Here are the screenshots of the prototype on which I am working

![mid1](https://cloud.githubusercontent.com/assets/16368427/26734836/1cc4ef9e-47dd-11e7-92f1-73572a6c18a1.png)

![mid2](https://cloud.githubusercontent.com/assets/16368427/26734875/44d51414-47dd-11e7-99cb-6494e84db185.png)

Work is still left, so i will be sending another PR for the middle section so that PRs do not become too large. The middle section will be able to render markdown.

djmgit commented:
@mariobehling since the present app.json files do not provide all the required information, so for now I have modified the app.json of whatsTrending app with dummy data for testing purpose. Once this page is ready, the app.json of the existing apps will be updated one by one, relevant issues will be created. The guidelines for creating a proper app.json file will be provided on the tutorial page which I will be creating soon.

kavithaenair commented on details.html:
```diff
@@ -0,0 +1,102 @@
+              <div class="sidebar-header-image">
+                <img src = "images/sidebar-images/menu.png">
+              </div>
+              <div class="catgegory-header">
```
Please fix the typos in the class name and give a appropriate name.

kavithaenair replied:
I cannot find a page header here where it displays the category of the apps as in apps.loklak.org .

djmgit replied:
@kavithaenair the category to which the selected app belongs will be highlighted in the left category sidebar

djmgit replied:
@kavithaenair the category to which the selected app belongs will be highlighted in the left category sidebar

kavithaenair replied:
@djmgit  correct the typo here as well `Know more about ths category` on the category navbar.

djmgit commented:
@kavithaenair that is the ultimate target, when a an user clicks on any app, it will go to the store listing page, there the user will come to know everything about the app and then he/she can try the app using the link provided (Try now)

djmgit commented:
Since the page is not ready right now and the app.json of all the apps are not ready so I have yet not integrated the storelisting page with all the apps, presently I am only using whatsTrending app for testing purpose (and dummy data as shown in the screenshots)

kavithaenair commented:
@djmgit while refreshing the page it loads slowly and also shows `{{category.name}}` in the categories navbar and `{{app.name}}` on the page (as images as loading slowly).

djmgit commented:
@kavithaenair I have removed 'know more about this category' as it was a place holder which I had forgotten to remove. 
I have moved the angularjs script to top of the page, hope this will prevent the angular variables from getting displayed on the screen. Regarding speed, I will try to optimize it more as the page develops.

mariobehling commented:
@djmgit Ok, thanks for explaining. So, this is work in progress then. With the screenshot that you are showing I see that the diagram is scaled out of proportion. This is also the case on the current page. I would expect that you see this yourself and correct it when you work on this. Please fix issues like that on the go. Cheers.

vibhcool commented on css/details.css:
```diff
@@ -0,0 +1,154 @@
+}
+
+.menu-list-item {
+    text-decoration: none !important;
```
Could you prevent using !important as it leads to more errors in future. 
see this -> https://stackoverflow.com/questions/9245353/what-does-important-in-css-mean

djmgit commented:
@vibhcool removed !important

Achint08 commented on app_list.js:
```diff
@@ -34,6 +42,9 @@ app.controller("app_list", function ($scope, $http) {
 
 
     $scope.categoryFilter = function (event) {
+        if ($scope.category !== null) {
```
Can't this condition be removed and let `$scope.category` be made null directly?

djmgit commented:
@Achint08 thanks for pointing out, removed the null checking condition
